local.c

```diff
@@ -1340,6 +1351,18 @@ static int add_attr_to_channel(struct iio_channel *chn,
 	attrs[chn->nb_attrs++].name = name;
 	chn->attrs = attrs;
 	DEBUG("Added attr \'%s\' to channel \'%s\'\n", name, chn->id);
+
+	for (i = chn->nb_attrs - 1; i > 0;  i--) {
+		if (strcmp(attrs[i].name, attrs[i-1].name) < 0) {
+			tmp1 = attrs[i].filename;
```

Comment:
for this case particularly, I'd use a `struct iio_channel_attr tmp;` and memcpy() stuff onto it and back with `sizeof(tmp)`
it's one variable, and can handle future cases [new members on the `struct iio_channel_attr ` ] ;

Reply:
as a continuation here [and while thinking about it a bit], an idea would be to do a search for the index/position where the new element could be added;
then after the `realloc` memmove() everything +1 position to the right;

so, something like:
```
int insert_idx = 0;
for (i = 0; i < chn->nb_attrs - 1;  i--) {
	if (strcmp(attrs[i].name, attrs[i+1].name) >= 0)
		break
}
insert_idx = i;  /* or just use `i` directly */

attrs = realloc(chn->attrs, (1 + chn->nb_attrs) * sizeof(struct iio_channel_attr));

memmove(attrs[i+1], attrs[i], chn->nb_attrs - i);
chn->nb_attrs++;
```

i don't have a strong preference about doing it either way;